examples/simple/igraph_get_shortest_paths_bellman_ford.c

```diff
@@ -0,0 +1,237 @@
+                u = v;
+                v = dummy;
+            }
+            if (v != i) {
+                exit(error_code + 5);
+            } else if (u != VECTOR(*pred)[i]) {
+                exit(error_code + 6);
+            }
+        }
+    }
+
+    return 0;
+}
+
+int main() {
+    igraph_t g;
+    igraph_vector_ptr_t vecs, evecs;
+    igraph_vector_long_t pred, inbound;
+    long int i;
+    igraph_real_t weights_data_0[] = { 0, 2, 1, 0, 5, 2, 1, 1, 0, 2, 2, 8, 1, 1, 3, 1, 1, 4, 2, 1 };
+    igraph_real_t weights_data_1[] = { 6, 7, 8, -4, -2, -3, 9, 2, 7 };
+    igraph_real_t weights_data_2[] = { 6, 7, 2, -4, -2, -3, 9, 2, 7 };
+    igraph_vector_t weights_vec;
+    igraph_vs_t vs;
+    igraph_vector_ptr_init(&vecs, 6);
+    igraph_vector_ptr_init(&evecs, 6);
+    igraph_vector_long_init(&pred, 0);
+    igraph_vector_long_init(&inbound, 0);
+
+    for (i = 0; i < igraph_vector_ptr_size(&vecs); i++) {
+        VECTOR(vecs)[i] = calloc(1, sizeof(igraph_vector_t));
+        igraph_vector_init(VECTOR(vecs)[i], 0);VECTOR(evecs)[i] = calloc(1, sizeof(igraph_vector_t));
+        igraph_vector_init(VECTOR(evecs)[i], 0);
+    }
+    igraph_vs_vector_small(&vs, 0, 1, 3, 5, 2, 1,  -1);
+    igraph_small(&g, 10, IGRAPH_DIRECTED,
+                0, 1, 0, 2, 0, 3,    1, 2, 1, 4, 1, 5,
+                2, 3, 2, 6,         3, 2, 3, 6,
+                4, 5, 4, 7,         5, 6, 5, 8, 5, 9,
+                7, 5, 7, 8,         8, 9,
+                5, 2,
+                2, 1,
+                -1);
+    igraph_vector_view(&weights_vec, weights_data_0, sizeof(weights_data_0) / sizeof(igraph_real_t));
+    igraph_get_shortest_paths_bellman_ford(&g, /*vertices=*/ &vecs,
+                                    /*edges=*/ &evecs, /*from=*/ 0, /*to=*/ vs,
+                                    &weights_vec, IGRAPH_OUT,
+                                    /*predecessors=*/ &pred,
+                                    /*inbound_edges=*/ &inbound);
+
+    check_evecs(&g, &vecs, &evecs, 30);
+    check_pred_inbound(&g, &pred, &inbound, /* from= */ 0, 60);
+
+    for (i = 0; i < igraph_vector_ptr_size(&vecs); i++) {
+            print_vector(VECTOR(vecs)[i]);
+            igraph_vector_destroy(VECTOR(vecs)[i]);
+            free(VECTOR(vecs)[i]);
+            igraph_vector_destroy(VECTOR(evecs)[i]);
+            free(VECTOR(evecs)[i]);
+    }
+
+    printf("\n");
+
+    /***************************************/
+
+    /* Graph with negative weights */
+
+    igraph_vector_ptr_init(&vecs, 5);
+    igraph_vector_ptr_init(&evecs, 5);
+    igraph_vector_long_init(&pred, 0);
+    igraph_vector_long_init(&inbound, 0);
+
+    for (i = 0; i < igraph_vector_ptr_size(&vecs); i++) {
+        VECTOR(vecs)[i] = calloc(1, sizeof(igraph_vector_t));
+        igraph_vector_init(VECTOR(vecs)[i], 0);VECTOR(evecs)[i] = calloc(1, sizeof(igraph_vector_t));
+        igraph_vector_init(VECTOR(evecs)[i], 0);
+    }
+    igraph_vs_vector_small(&vs, 0, 1, 3, 2, 1,  -1);
```

Comment:
Another leak is here: this re-allocates on top of the original `vs` without destroying it first.

Reply:
added destroy function to vs